### ISSUE TITLE
Random TCP timeouts when accessing a service (br_netfilter unloaded? )

### ISSUE THREAD

**Issue by TattiQ** [kubernetes/kops#5393]
1. What `kops` version are you running? The command `kops version`, will display
 this information. 1.8

2. What Kubernetes version are you running? `kubectl version` will print the
 version if a cluster is running or provide the Kubernetes version specified as
 a `kops` flag. 1.7.12 

3. What cloud provider are you using? aws

4. What commands did you run?  What is the simplest way to reproduce this issue?
It turns out the br_netfilter module was not loaded on a  kops node which caused random  timeouts when one pod was trying to access a service. Both pods were residing on the same node. 

from the source pod's logs it looked like this: 

ConnectTimeout(MaxRetryError("HTTPConnectionPool(host='destination_svc', port=80): Max retries exceeded with url: /tasks  (Caused by ConnectTimeoutError(<urllib3.connection.HTTPConnection object at 0x7f538247580>, 'Connection to destination_svc timed out. (connect timeout=15)'))",),) 

Service  was passing the request to the endpoint (pod cluster ip) but then there was a tcp reset.  

5. What happened after the commands executed?

6. What did you expect to happen?
I expect the br_netfilter to be loaded and some config management to make sure it stays loaded. 
We use flannel(vxlan) and cni plugin. Which part is responsible for loading the kernel modules? 
br_netfilter is a separate module since 3.18 kernel .  All the other nodes did have the module loaded, so it happens at some point, but how could it be unloaded then (nothing particular in the node's logs regarding that) and how can we make sure the config on kops nodes is consistent? 

7. Please provide your cluster manifest. Execute
  `kops get --name my.example.com -o yaml` to display your cluster manifest.
  You may want to remove your cluster name and other sensitive information.

8. Please run the commands with most verbose logging by adding the `-v 10` flag.
  this looks related ->  https://github.com/kubernetes/kops/issues/4323 . 
 

9. Anything else do we need to know?
I noticed some people add a hook to the  kops cluster config that loads the module . It is the way to do it ? Anything regarding that in kops documentation that I missed? If so, please point me to it. 
Since the br_netfilter has been loaded we do not experience this issues. 

Thank you!  

------------- FEATURE REQUEST TEMPLATE --------------------

1. Describe IN DETAIL the feature/behavior/change you would like to see.

2. Feel free to provide a design supporting your feature request.

**Comment by schweizerbolzonello** [comment]
@TattiQ 

What AMI are you using for the nodes ? CoreOS ?

**Comment by TattiQ** [comment]
Hi @schweizerbolzonello , sorry, I should've added that info, it's k8s-1.7-debian-jessie-amd64-hvm-ebs-2018-01-05 - ami-59821420
Kubernetes 1.7 Base Image - Debian jessie amd64 .

**Comment by ms4720** [comment]
I had br_netfilter unloaded/turned off on ubuntu 16.04 lts and things just did not start using k8s 1.10 or 1.11, forgot which

**Comment by TattiQ** [comment]
HI guys, experienced it again on a different kops cluster, also Debian jessie amd64 , kops 1.7.16.

**Comment by TattiQ** [comment]
And yet again today in prod, experienced dns resolution issues due to the same issue. from k8s perspective the node in question was 22 days , as kubectl get nodes shows, but it's uptime was a couple of hours, so the node was rebooted for some reason and br_netfilter was not loaded. 

log//kern.log:Aug  8 07:11:36 ip- kernel: [   11.947780] ip_tables: (C) 2000-2006 Netfilter Core Team
log//kern.log:Aug  8 07:11:43 ip- kernel: [   19.385788] bridge: automatic filtering via arp/ip/ip6tables has been deprecated. Update your scripts to load br_netfilter if you need this.
log//kern.log:Aug  8 07:11:50 ip- kernel: [   26.761103] Netfilter messages via NETLINK v0.30.

As I see there is a PR related to fixing this in newer versions https://github.com/kubernetes/kops/pull/5490 . 
but what about mere mortals who  run 1.7.X kubernetes...

**Comment by ms4720** [comment]
Do you force load netfilter on boot?

**Comment by TattiQ** [comment]
yep! via a oneshot systemd unit. as I understood they plan to do the same here but by adding the module to a file under /etc/modules-load.d/   https://github.com/kubernetes/kops/pull/5490/files .  e.g. https://www.freedesktop.org/software/systemd/man/modules-load.d.html
